E-Hospital/E-Hospital.BusinessLogic/Services/Implementation/UserService.cs

```diff
@@ -53,6 +57,10 @@ public void ChangeRequestState(int visitRequestId, bool isApproved)
             request.IsApproved = isApproved;
 
             _requestsRepository.Update(request);
+
+            var selectedPatient = _activePatients.FirstOrDefault(x => x.Key.Id == request.Doctor.Id);
```

Comment:
We don't need compare Id of patient with DoctorId. Instead compare with `request.Patient.Id`.

Reply:
Also, as you want to access Doctor's or Patient's Id, you need to include them in the 55 line. Without that, you will have null-exception.
Or compare `request.DoctorId` instead of `request.Doctor.Id` and you wouldn't need to include any additional properties.